robot/client/client_session_test.go

```diff
@@ -101,8 +102,12 @@ func TestClientSessionOptions(t *testing.T) {
 
 						sessMgr := &sessionManager{}
 						arbName := resource.NewName(echoAPI, "woo")
+						injectResources := []resource.Name{arbName}
 						injectRobot := &inject.Robot{
-							ResourceNamesFunc: func() []resource.Name { return []resource.Name{arbName} },
+							ResourceNamesFunc: func() []resource.Name { return injectResources },
+							MachineStatusFunc: func(ctx context.Context) (robot.MachineStatus, error) {
+								return rdktestutils.ResourcesToMachineStatus(injectResources), nil
```

Comment:
the bulk of test updates in this PR involve injecting `MachineStatus` function to test robot clients, since this API replaces `ResourceNames` as the means of sourcing resources